src/NzbDrone.Core/Download/Clients/Hadouken/HadoukenProxy.cs

```diff
@@ -0,0 +1,116 @@
+using NzbDrone.Core.Rest;
+using RestSharp;
+
+namespace NzbDrone.Core.Download.Clients.Hadouken
+{
+    public sealed class HadoukenProxy : IHadoukenProxy
+    {
+        private static int _callId;
+        private readonly Logger _logger;
+
+        public HadoukenProxy(Logger logger)
+        {
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+
+            _logger = logger;
+        }
+
+        public HadoukenSystemInfo GetSystemInfo(HadoukenSettings settings)
+        {
+            return ProcessRequest<HadoukenSystemInfo>(settings, "core.getSystemInfo").Result;
+        }
+
+        public IDictionary<string, HadoukenTorrent> GetTorrents(HadoukenSettings settings)
+        {
+            return ProcessRequest<Dictionary<string, HadoukenTorrent>>(settings, "session.getTorrents").Result;
+        }
+
+        public IDictionary<string, object> GetConfig(HadoukenSettings settings)
+        {
+            return ProcessRequest<IDictionary<string, object>>(settings, "config.get").Result;
+        }
+
+        public string AddTorrentFile(HadoukenSettings settings, byte[] fileContent)
+        {
+            return ProcessRequest<string>(settings, "session.addTorrentFile", Convert.ToBoolean(fileContent)).Result;
+        }
+
+        public void AddTorrentUri(HadoukenSettings settings, string torrentUrl)
+        {
+            ProcessRequest<string>(settings, "session.addTorrentUri", torrentUrl);
+        }
+
+        public void RemoveTorrent(HadoukenSettings settings, string downloadId, bool deleteData)
+        {
+            ProcessRequest<bool>(settings, "session.removeTorrent", downloadId, deleteData);
+        }
+
+        private HadoukenResponse<TResult> ProcessRequest<TResult>(HadoukenSettings settings,
+            string method,
+            params object[] parameters)
+        {
+            var client = BuildClient(settings);
+            return ProcessRequest<TResult>(client, method, parameters);
+        }
+
+        private HadoukenResponse<TResult> ProcessRequest<TResult>(IRestClient client,
+            string method,
+            params object[] parameters)
+        {
+            var request = new RestRequest(Method.POST);
+            request.Resource = "api";
+            request.RequestFormat = DataFormat.Json;
+            request.AddHeader("Accept-Encoding", "gzip,deflate");
+
+            var data = new Dictionary<String, Object>();
+            data.Add("id", GetCallId());
+            data.Add("method", method);
+
+            if (parameters != null)
+            {
+                data.Add("params", parameters);
+            }
+
+            request.AddBody(data);
+
+            _logger.Debug("Url: {0} Method: {1}", client.BuildUri(request), method);
+            return client.ExecuteAndValidate<HadoukenResponse<TResult>>(request);
+        }
+
+        private IRestClient BuildClient(HadoukenSettings settings)
```

Comment:
Use https://github.com/Sonarr/Sonarr/blob/develop/src/NzbDrone.Common/Http/HttpClient.cs instead of RestClient.